services/hh/processor.go

```diff
@@ -144,8 +157,25 @@ func (p *Processor) Process() error {
 	p.mu.RLock()
 	defer p.mu.RUnlock()
 
-	res := make(chan error, len(p.queues))
-	for nodeID, q := range p.queues {
+	// Only process queues for active nodes.
+	var activeQueues map[uint64]*queue
+	if p.MetaProvider == nil {
+		// No MetaProvider so process all queues.
+		activeQueues = p.queues
+	} else {
+		activeQueues = make(map[uint64]*queue)
+		activeNodes := p.MetaProvider.Nodes()
+		for nodeID, q := range p.queues {
+			if p.MetaProvider != nil && !contains(activeNodes, nodeID) {
```

Comment:
we are in the else block where we already checked for MetaProvider not being nil.  Is this due to another process shutting down the meta store?